### PR TITLE
[Feature][AES-1361] Build Content Hub Article List Page

### PR DESCRIPTION
**Pull request by zack76**
#### Ticket No.
AES-1361

#### Link to JIRA Ticket if applicable
https://aesoponline.atlassian.net/browse/AES-1361

#### Issue / Feature description
Build content hub related components in Aesop-Gel

#### Development sumary
New components: 

- ContentHubArticle
Screen Shot 2020-07-20 at 3 29 37 pm

- ContentHubArticleList
Screen Shot 2020-07-20 at 3 30 20 pm

- ReadMore
Screen Shot 2020-07-20 at 3 30 56 pm

- LoadMoreButton
Screen Shot 2020-07-20 at 3 28 44 pm
Screen Shot 2020-07-20 at 3 28 46 pm

### CODE REVIEW

File: src/components/ContentHubArticle/ContentHubArticle.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  isReadMore, // if it's displayed in ReadMore section
+  isMenuItem, // if it's displayed in Read Menu Navigation
+  title,
+  readingTime,
+  category,
+  className,
+  uri,
+}) {
+  const classSet = cx(styles.base, className);
+
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.js:25]: Sorry to be pedantic, but can we tighten up these constant declarations? 

Also, order them with hooks at top, then general constant declarations and classSets at the bottom. 

As hooks should always be at the top as they cause side effects, general constants rely on the results of hooks at times, and classSets rely on the computation of all of the above.

``` 
const ref = useRef(null);
const isOnScreen = useOnScreen(ref, 0.4);
const currentImage = isHorizontal ? horizontalThumbnail : verticalThumbnail;
const classSet = cx(styles.base, className);
const categoryClassSet = cx(styles.category);
const imageClassSet = cx(styles.link);
const noteClassSet = cx(styles.note);
const titleClassSet = cx(styles.title, styles.link);
```

File: src/components/ContentHubArticle/ContentHubArticle.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import cx from 'classnames';
+import Transition from '~/components/Transition';
+import Image from '~/components/Image';
+import Heading from '~/components/Heading';
+import Hyperlink from '~/components/Hyperlink';
+import styles from './ContentHubArticle.module.css';
+import { useOnScreen } from '~/customHooks/useOnScreen';
+
+const ContentHubArticle = forwardRef(function ContentHubArticleRef({
+  id,
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.js:12]: make sure object declarations and object destructuring is alphabetically sorted (as with props) for easier scan reading

Comment (beau-charman) [reply src/components/ContentHubArticle/ContentHubArticle.js:12]:
```
category,
className,
horizontalThumbnail,
isHorizontal,
isMenuItem,
isReadMore, 
readingTime,
title,
uri,
verticalThumbnail,
id,
```

File: src/components/ContentHubArticle/ContentHubArticle.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
+import React, { forwardRef, useRef } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import Transition from '~/components/Transition';
+import Image from '~/components/Image';
+import Heading from '~/components/Heading';
+import Hyperlink from '~/components/Hyperlink';
+import styles from './ContentHubArticle.module.css';
+import { useOnScreen } from '~/customHooks/useOnScreen';
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.js:9]: With imports, order them as modules first (with react at the top) then application level imports (constants, utils, contexts, hooks, then components) and then current component level requirements (styles etc...), in alphabetical order too for scan reading. 

```
import React, { forwardRef, useRef } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { useOnScreen } from '~/customHooks/useOnScreen';
import Image from '~/components/Image';
import Heading from '~/components/Heading';
import Hyperlink from '~/components/Hyperlink';
import Transition from '~/components/Transition';
import styles from './ContentHubArticle.module.css';
```

File: src/components/ContentHubArticle/ContentHubArticle.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  const ref = useRef(null);
+  const isOnScreen = useOnScreen(ref, 0.4);
+
+  return (
+    <Transition isActive={isOnScreen} type="fade">
+      <div className={classSet} id={id} ref={ref}>
+        {isReadMore && !isMenuItem && (
+          <Hyperlink
+            className={imageClassSet}
+            title={`Link to ${title}`}
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.js:45]: I know I have done it in the past, but we will keep out any untranslated copy. just leave the title here as `title`, remove the 'Link to ' english.

File: src/components/ContentHubArticle/ContentHubArticle.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+              small={currentImage.small}
+            />
+          </Hyperlink>
+        )}
+        {!isMenuItem && (
+          <Heading className={categoryClassSet} level={'3'} size="xSmall">
+            {category}
+          </Heading>
+        )}
+        <Hyperlink title={`Link to ${title}`} url={uri}>
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.js:62]: Same here with the english copy

File: src/components/ContentHubArticle/ContentHubArticle.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+              altText={currentImage.altText}
+              className={styles.image}
+              large={currentImage.large}
+              medium={currentImage.medium}
+              small={currentImage.small}
+            />
+          </Hyperlink>
+        )}
+        {!isMenuItem && (
+          <Heading className={categoryClassSet} level={'3'} size="xSmall">
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.js:58]: Not sure why you didn't get a prop type error here, but level is a string. no need for a number type as there is no computations happing.

File: src/components/ContentHubArticle/ContentHubArticle.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+  verticalThumbnail: PropTypes.object,
+  horizontalThumbnail: PropTypes.object,
+  isHorizontal: PropTypes.bool,
+  isReadMore: PropTypes.bool,
+  isMenuItem: PropTypes.bool,
+  title: PropTypes.string,
+  readingTime: PropTypes.string,
+  category: PropTypes.string,
+  className: PropTypes.string,
+  uri: PropTypes.string,
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.js:102]: Alphabetize these and the defaultProps. Thank you!

File: src/components/ContentHubArticle/ContentHubArticle.module.css [hunk 8]
```diff
@@ -0,0 +1,43 @@
+@import 'styles/index';
+
+.base {
+  display: block;
+  padding: 15px;
+}
+
+.category {
+  /* padding: 10px 10px 10px 0; */
+  border-bottom: 1px solid #bcbbb4;
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.module.css:10]: We got a variable for `#bcbbb4`?

File: src/components/ContentHubArticle/ContentHubArticle.module.css [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  font-size: 16px;
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.module.css:11]: Wrap `font-size` value in the `rem` function, so later on we can use responsive typography. 

```
font-size: rem(16px);
```

File: src/components/ContentHubArticle/ContentHubArticle.module.css [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+  }
+}
+
+.title {
+  border-bottom: 1px solid transparent;
+  color: var(--color-grey-20);
+  font-family: var(--font-family-zapf-serif);
+  font-size: rem(30px);
+  letter-spacing: 0;
+  line-height: 37px;
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.module.css:33]: Use unit-less values for line-heights, especially fixed pixel values. Unit-less values computate and scales the line-heights more effectively for response typography. 

Not sure what the value should be here, based on the 30px font-size, an example would be: 

```
line-height: 1.8;
```

File: src/components/ContentHubArticle/ContentHubArticle.module.css [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+  transition: all 500ms ease;
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.module.css:34]: Try and use the easings.css file to store transition-animation properties

File: src/components/ContentHubArticle/ContentHubArticle.module.css [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+.title:hover {
+  border-bottom: 1px solid #333;
```
Comment (beau-charman) [inline src/components/ContentHubArticle/ContentHubArticle.module.css:38]: Grab the variable for the #333 hex.

File: src/components/ContentHubArticleList/ContentHubArticleList.module.css [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    padding-right: 35px;
+    padding-left: 35px;
+  }
+}
+
+.article {
+  margin-bottom: 1rem;
+}
+
+.topLeft-0 {
```
Comment (beau-charman) [inline src/components/ContentHubArticleList/ContentHubArticleList.module.css:24]: Not sure why youre not getting an error here for the camel case

File: src/components/LoadMoreButton/LoadMoreButton.js [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+    console.error('updateError'); // eslint-disable-line
+  }
+
+  return (
+    <Button
+      className={classSet}
+      isAlternate={true}
+      isEnabled={!isLoading && isEnabled}
+      onClick={handleOnClick}
+      theme={'light'}
```
Comment (beau-charman) [inline src/components/LoadMoreButton/LoadMoreButton.js:42]: Just put string literal props in double quotes: `theme="light"`

File: src/components/LoadMoreButton/LoadMoreButton.module.css [hunk 15]
```diff
@@ -0,0 +1,66 @@
+@import 'styles/index';
+
+.base {
+  min-width: auto;
+  max-width: 100%;
+  border: 1px solid var(--color-grey-60) !important;
```
Comment (beau-charman) [inline src/components/LoadMoreButton/LoadMoreButton.module.css:6]: Why we need an important on a base style? I understand if it was for a hover effect or a side effect, but what is happening here?

File: src/components/ReadMore/ReadMore.js [hunk 16]
```diff
@@ -0,0 +1,44 @@
+import React, { forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import ContentHubArticle from '~/components/ContentHubArticle';
+import Transition from '~/components/Transition';
+import styles from './ReadMore.module.css';
+import { ascertainIsSmallOrMediumOnlyViewport } from '~/utils/viewports';
```
Comment (beau-charman) [inline src/components/ReadMore/ReadMore.js:7]: Move this below the cx import

**Comment by beau-charman** [comment]
Also, no need to rebuild the build files as they are automatically built in the deployment process

File: src/components/LoadMoreButton/LoadMoreButton.module.css (hunk 15 above)
Comment (zack76) [reply src/components/LoadMoreButton/LoadMoreButton.module.css:6]: > Why we need an important on a base style? I understand if it was for a hover effect or a side effect, but what is happening here?

the value of 'isAlternate' for this button is true, so it has defined the border color which is not same as the design for loadmore button. 

Without important: No grey border
Screen Shot 2020-07-23 at 2 50 54 pm

With important: grey border
Screen Shot 2020-07-23 at 2 51 00 pm

**Pull request merged**